octoprint_mrbeam/analytics/review_handler.py

```diff
@@ -48,11 +49,28 @@ def _on_mrbeam_plugin_initialized(self, event, payload):
         ReviewFileUploader.upload_now(self._plugin, self._review_lock)
 
     def save_review_data(self, data):
-        self._write_review_to_file(data)
-        self._settings.set_boolean(["review", "given"], data["dontShowAgain"])
-        self._settings.save()  # This is necessary because without it the value is not saved
-
-        ReviewFileUploader.upload_now(self._plugin, self._review_lock)
+        given = self._settings.get(["review", "given"])
+        if not given or data.get("debug", False):
+            data = self._add_review_data(data)
+            self._write_review_to_file(data)
+            self._settings.set_boolean(["review", "given"], data["dontShowAgain"])
+            self._settings.save()  # This is necessary because without it the value is not saved
+
+            ReviewFileUploader.upload_now(self._plugin, self._review_lock)
+        else:
+            self._logger.warn("Not accepting user review since it was already given.")
```

Comment:
How about updating the dataset with `data["already_given"] = given` and filter afterwards, perhaps to see if the review changed

Reply:
Each device should only send one review ever (only exception: if the device got reset). So far this was only enforced on the frontend side while the backend accepted happily as many reviews as the frontend (or a dev tools hacker) submitted. This is just to enforce this policy on the backend as well. Therefor I thinks it's ok this way.